backend/src/apiserver/server/list_request_util.go

```diff
@@ -153,7 +153,7 @@ func parseAPIFilter(encoded string) (*api.Filter, error) {
 	return f, nil
 }
 
-func validatedListOptions(listable list.Listable, pageToken string, pageSize int, sortBy string, filterSpec string) (*list.Options, error) {
+func ValidatedListOptions(listable list.Listable, pageToken string, pageSize int, sortBy string, filterSpec string) (*list.Options, error) {
```

Comment:
I think we should think a bit more about how we want to structure and reuse components from v1.

- Because v2beta is a separate directory under `/src/apiserver`, I am concerned that it gives a false impression that files outside of the directory is not used
- If we start using the v1 component for v2 without some structure in place, I think it would complicate v1 clean-up in the future.

Would it make more sense to duplicate the file under `/src/apiserver/v2beta1/server` or introduce a common directory between the two APIs?

Reply:
With differences in packages `v2beta1` and `server`, we should look into the best way to handle the deprecation of `server` in the future

Reply:
Yes, we need to think about how to handle the common code appropriately. 
I am not in favour of duplicating code.. as if there is a bug or need of enhancement, it should be done in both places. And if a bug is opened for just one of the versions - the developer may not realize that the fix should be done in two places.  In addition, unit tests should be also duplicated.

- So, I am more in favour of making v2 dependent on some common exported v1.funcs, but the opposit should be forbidden. Once it come the time to remote v1, IMO if I delete  v1.func that is used in v2 - I will have compilation errors -> so I will keep just what I need or I could insperts my imports.  

- Or to make a restructure -  create a dedicated folder for v1
```
|> server
   |> v1
   |> v2beta1 
   | utils.go
   | utils_test.go
   | common.go
   | common_test.go
```  
WDYT?

Reply:
Fr funcs `validatedListOptions` why not get used of the [common](https://github.com/difince/pipelines/pull/3/commits/9482f5053275d560df9efd8063c0f01a638ed4a0) package that already exists

Reply:
[Commit ](https://github.com/difince/pipelines/pull/3/commits/f5b501c908729a2ebad0c659e0d9a8acd1a8bd35) moves `validatedListOptions` into the `list` package.